routing/manager/manager.go

```diff
@@ -0,0 +1,601 @@
+// Copyright (c) 2016 Bitfury Group Limited
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php
+
+package manager
+
+import (
+	"errors"
+	"fmt"
+	"github.com/lightningnetwork/lnd/routing/pbuffer"
+	"github.com/lightningnetwork/lnd/routing/rt"
+	"github.com/lightningnetwork/lnd/routing/rt/graph"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"time"
+)
+
+var (
+	TimeoutError      = errors.New("Operation timeout")
+	PathNotFoundError = errors.New("Path not found")
+)
+
+type RoutingMessage struct {
+	SenderID   graph.ID
+	ReceiverID graph.ID
+	Msg        lnwire.Message
+}
+
+type addChannelCmdMessage struct {
+	Id1, Id2 graph.ID
+	TxID     graph.EdgeID
+	Info     *graph.ChannelInfo
+	err      chan error
+}
+
+type removeChannelCmdMessage struct {
+	Id1, Id2 graph.ID
+	TxID     graph.EdgeID
+	err      chan error
+}
+
+type hasChannelCmdMessage struct {
+	Id1, Id2 graph.ID
+	TxID     graph.EdgeID
+	rez      chan bool
+	err      chan error
+}
+
+type openChannelCmdMessage struct {
```

Comment:
addChannel simply adds a channel to the routing table. It does not inform neighbors, ...
openChannel is used when a new neighbor is added. It will trigger sending NeighborHello, update for other neighbors, ...